depmapomics/fingerprinting.py

```diff
@@ -0,0 +1,423 @@
+
+
+async def fingerPrint(
+    rnasamples,
+    wgssamples,
+    sid='id',
+    sampleset=SAMPLESETNAME,
+    allbatchpairset=FPALLBATCHPAIRSETS,
+    workspace=FPWORKSPACE,
+    working_dir=WORKING_DIR,
+    bamcolname=LEGACY_BAM_COLNAMES,
+    prev_mat_df=None,
+    updated_mat_filename=""
+):
+    """1.1  Generate Fingerprint VCFs
+
+    Here we use Dalmatian to run the fingerprint_bam_with_liftover workflow on Terra.
+    This workflow calls Picard ExtractFingerprint to generate a fingerprint VCF and 
+    then calls Picard LiftoverVcf to covert this vcf to hg38. To fingerprint hg38 bam files 
+    just run fingerprint_bam instead.
+
+    Args:
+        samples ([type]): [description]
+        sampleset ([type]): [description]
+        sid ([type]): [description]
+        working_dir ([type]): [description]
+        bamcolname ([type]): [description]
+        workspace ([type], optional): [description]. Defaults to WORKSPACE.
+
+    Author:
+        William Colgan (wcolgan@broadinstitute.org)
+    """
+
+    samples = pd.concat([rnasamples, wgssamples])
+    bams = samples[bamcolname]
+    bams[sid] = bams.index
+    print("adding " + str(len(bams)) + " new samples to the fingerprint")
+    wm = dm.WorkspaceManager(workspace).disable_hound()
+
+    # Upload sample sheet
+    samples_df = pd.DataFrame(
+        bams[bamcolname + [sid, sid]].values,
+        columns=["bam_filepath", "bai_filepath", "sample_id", "participant_id"],
+    )
+    samples_df = samples_df.set_index("sample_id")
+    wm.upload_samples(samples_df, add_participant_samples=True)
+    wm.update_sample_set(sampleset, samples_df.index)
+    add_sample_batch_pairs(wm, working_dir=WORKING_DIR)
+
+    # Submit fingerprinting jobs, generate vcf files for all lines
+    submission_id = wm.create_submission(
+        "fingerprint_bam_with_liftover",
+        sampleset,
+        "sample_set",
+        expression="this.samples",
+    )
+    await terra.waitForSubmission(workspace, submission_id)
+
+    # 1.2  Crosscheck Fingerprint VCFs
+    # Here we use Dalmation to run the crosscheck_vcfs workflow on Terra.
+    # This workflow calls Picard CrosscheckFingerprints to compare vcfs between every
+    # sample_batch_pair in the sample_batch_pair_set
+
+    # Submit crosscheck jobs
+    conf = wm.get_config("crosscheck_vcfs")
+    wm.update_config(conf)
+    submission_id = wm.create_submission(
+        "crosscheck_vcfs",
+        allbatchpairset,
+        "sample_batch_pair_set",
+        expression="this.sample_batch_pairs",
+    )
+    await terra.waitForSubmission(workspace, submission_id)
+
+    # save config after jobs finish running
+    terra.saveWorkspace(workspace, "data/" + sampleset + "/FPconfig/")
+
+    # Update LOD matrix
+    new_ids, updated_lod_mat = updateLOD(wm, sampleset, working_dir, save_new_mat=False, prev_mat_df=None, updated_mat_filename=updated_mat_filename)
+
+    # finding issues with the dataset
+    v = updated_lod_mat.loc[new_ids]
```

Comment:
change v too?

Reply:
just to clarify, should the input for checkMismatches() and checkMatches() be just the portion of the lod matrix with the new lines? I think in 21Q4 v wasn't defined in the notebook so I assumed it was the whole matrix, but it makes more sense if it's just the slice with new ids